agents/src/voice/audio_recognition.ts

```diff
@@ -37,7 +37,9 @@ export class AudioRecognition {
 
   start() {
     this.vadStreamProcessor = this.vadTask().catch((err) => {
-      this.logger.error('Error in VAD task', err);
+      // raise the error
+      throw err;
```

Comment:
this will always throw as an unhandled exception, which we'd want to avoid. 

You're trying to handle an error in an async task from within a non-async method. I think `catch` is the best you can do there